### ISSUE TITLE
Infracost comment github --tag

### ISSUE THREAD

**Issue by Laucans** [infracost/actions#60]
Hi, 

# Context
Got a github repository which manage 2 environment, prod and non-prod. Theses environment are managed by github-action through matrix.

Infracost setup : 
```
      - name: Setup Infracost
        uses: infracost/actions/setup@v1
```
Got an issue with the following command
```bash
infracost comment github --path /tmp/infracost-${{ matrix.environment }}.json \
                                   --repo $GITHUB_REPOSITORY \
                                   --github-token ${{github.token}} \
                                   --pull-request ${{github.event.pull_request.number}} \
                                   --behavior update \
                                   --tag "${{ matrix.environment }}"
```

# Issue
The behaviour of `--tag` seems wrong. 
Because of matrix job is executed twice. 
- The first one for example non-prod create a new comment which containing the tag "non-prod"
- The second one, prod, update the comment created by non-prod and replace the tag

# Expected
- The first one non-prod create a new comment which contains the tag "non-prod"
- The second one, prod, create a new comment  which contains the tag "prod"

**Comment by alikhajeh1** [comment]
@Laucans thanks for creating this issue! Can you please try using the tags "prod" and "dev" to see if that fixes the issue? I suspect the issue here is that "prod" is a substring of "non-prod" so the [matching logic](https://github.com/infracost/infracost/blob/master/internal/comment/github.go#L225-L227) is not properly differentiating between them.

**Comment by Laucans** [comment]
@alikhajeh1 thanks for answer.

So I tested with dev and prod and got the same behavior :
![image](https://user-images.githubusercontent.com/97902739/164231986-33f86e46-4377-405c-97c3-9cb71779a176.png)

I can't show the tag when it has been updated but if I change for behavior new :

![image](https://user-images.githubusercontent.com/97902739/164233282-030ac88e-be76-4ef9-ba04-e953e6d2b629.png)

![image](https://user-images.githubusercontent.com/97902739/164233328-d12888d7-76db-4f35-9dee-03b12ff3b5c6.png)

**Comment by Laucans** [comment]
Oh wait, the matching logic take the whole message body, dev is in my project name..

**Comment by Laucans** [comment]
Ok I changed dev for development and it's working. 

thanks :) 

Thank you.

**Comment by tim775** [comment]
This is fixed in 0.9.23
